common/services/deviceOperation.js

```diff
@@ -1,230 +1,209 @@
-import { BASE_URL, CERT } from '../constants'
+// noinspection JSUnusedGlobalSymbols,JSCheckFunctionSignatures
 
-import axios from 'axios'
-import doordeck from '../doordeck'
-import ephemaralKeyGenerator from './ephemeralKeyGenerator'
-
-var baseUrl = BASE_URL
-
-var signer = function (deviceId, operation) {
-  var privateKey = ephemaralKeyGenerator.retrieveSavedKeys().privateKey
+import axios from "axios";
+import jwtDecode from "jwt-decode";
+import KJUR from "jsrsasign";
 
+const signer = function (deviceId, operation) {
+  const privateKeyHex = KJUR.b64utohex(localStorage.privateKey);
+  const privateKey = KJUR.KEYUTIL.getKeyFromPlainPrivatePKCS8Hex(privateKeyHex);
   // Header
-  var oHeader = { alg: 'EdDSA', typ: 'JWT', x5c: JSON.parse(localStorage[CERT]).certificateChain }
+  const oHeader = {alg: "RS256", typ: "JWT"};
```

Comment:
RS256 is wrong, it should be EdDSA

Reply:
It looks like this library doesn't support Ed25519 which is fine, you can do signing using another library - https://github.com/kjur/jsrsasign/issues/341

You should be able to use this library to parse the X509 certificate to get the expiration date though, the next certificate in the list does use RSA